app/assets/javascripts/surveys.js

```diff
@@ -194,15 +202,29 @@
       userSurveys.trackEvent(survey.identifier, 'banner_shown', 'Banner has been shown')
     },
 
+    initialiseModuleOnElement: function (element, Module) {
+      if (Module) {
+        var module = new Module(element)
+        module.init()
+        return module
+      }
+    },
+
     displayURLSurvey: function (survey, surveyContainer) {
       var urlSurveyTemplate = userSurveys.getUrlSurveyTemplate()
       surveyContainer.innerHTML = urlSurveyTemplate.render(survey)
+      this.initialiseModuleOnElement(surveyContainer, window.GOVUK.Modules.Ga4LinkTracker)
+      this.initialiseModuleOnElement(surveyContainer, window.GOVUK.Modules.Ga4EventTracker)
+      this.initialiseModuleOnElement(surveyContainer.querySelector('[data-ga4-auto]'), window.GOVUK.Modules.Ga4AutoTracker)
       userSurveys.setURLSurveyEventHandlers(survey)
     },
 
     displayEmailSurvey: function (survey, surveyContainer) {
       var emailSurveyTemplate = userSurveys.getEmailSurveyTemplate()
       surveyContainer.innerHTML = emailSurveyTemplate.render(survey)
+      this.initialiseModuleOnElement(surveyContainer, window.GOVUK.Modules.Ga4LinkTracker)
+      this.initialiseModuleOnElement(surveyContainer, window.GOVUK.Modules.Ga4EventTracker)
+      this.initialiseModuleOnElement(surveyContainer.querySelector('[data-ga4-auto]'), window.GOVUK.Modules.Ga4AutoTracker)
```

Comment:
Looks like these three lines are repeated in both this function and `displayURLSurvey` - would it be sensible to move them into a separate function and call that?

Reply:
Will do :+1: The call to the GA4LinkTracker is different across both places, so I've had to account for that!